src/middlewared/middlewared/plugins/datastore.py

```diff
@@ -1,74 +1,160 @@
-from middlewared.service import CallError, Service, ValidationErrors
-from middlewared.schema import accepts, Any, Bool, Dict, Int, List, Ref, Str
-from sqlite3 import OperationalError
-
-import os
-import sys
-from itertools import chain
+from collections import defaultdict
+from concurrent.futures import ThreadPoolExecutor
+import operator
+import re
 
-sys.path.append('/usr/local/www')
-os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'freenasUI.settings')
+from sqlalchemy import and_, create_engine, func, inspect, select, types
+from sqlalchemy.sql import Alias
 
-import django
-from django.apps import apps
-if not apps.ready:
-    django.setup()
+from middlewared.schema import accepts, Any, Bool, Dict, Int, List, Ref, Str
+from middlewared.service import CallError, private, Service
+from middlewared.sqlalchemy import Model
+from middlewared.service_exception import MatchNotFound
 
-from django.apps import apps
-from django.db import connection
-from django.db.models import Q
-from django.db.models.fields.related import ForeignKey, ManyToManyField
+from middlewared.plugins.config import FREENAS_DATABASE
 
-# FIXME: django sqlite3_ha backend uses a thread to sync queries to the
-# standby node. That does not play very well with gevent+middleware
-# if that "thread" is still running and the originating connection closes.
-from freenasUI.freeadmin.sqlite3_ha import base as sqlite3_ha_base
-sqlite3_ha_base.execute_sync = True
 
-from middlewared.utils import django_modelobj_serialize
-from middlewared.service_exception import MatchNotFound
+def regexp(expr, item):
+    reg = re.compile(expr, re.I)
+    return reg.search(item) is not None
 
 
 class DatastoreService(Service):
 
     class Config:
         private = True
 
-    def _filters_to_queryset(self, filters, field_prefix=None):
+    thread_pool = ThreadPoolExecutor(1)
```

Comment:
Why not thread_pool within Config, so its done automatically?

Reply:
I don't want to run all service methods in this thread pool, it's just variable with same name.